File: deepseek_gemini_cli.js
#!/usr/bin/env node
import readline from 'node:readline';
import inquirer from 'inquirer';
import chalk from 'chalk';
import 'dotenv/config';

const API_URL = process.env.DEEPSEEK_API_ENDPOINT || 'https://api.deepseek.com/v1';
const API_KEY = process.env.DEEPSEEK_API_KEY;

if (!API_KEY) {
  console.error('Missing DEEPSEEK_API_KEY environment variable.');
  process.exit(1);
}

async function sendMessage(model, messages) {
  const res = await fetch(`${API_URL}/chat/completions`, {
    method: 'POST',
    headers: {
      'Content-Type': 'application/json',
      'Authorization': `Bearer ${API_KEY}`,
    },
    body: JSON.stringify({ model, messages }),
  });

  if (!res.ok) {
    throw new Error(`API error: ${res.status} ${await res.text()}`);
  }

  const data = await res.json();
  return data.choices[0].message.content;
}

async function chat(model) {
  const rl = readline.createInterface({ input: process.stdin, output: process.stdout });
  const history = [];

  console.log(chalk.green(`\nModel: ${model}\n`));

  function ask() {
    rl.question(chalk.blue('You: '), async (message) => {
      if (message.toLowerCase() === 'exit') {
        rl.close();
        return;
      }

      history.push({ role: 'user', content: message });

      try {
        const reply = await sendMessage(model, [
          { role: 'system', content: 'You are a helpful assistant.' },
          ...history,
        ]);
        console.log(chalk.yellow(`\nAI: ${reply}\n`));
        history.push({ role: 'assistant', content: reply });
      } catch (err) {
        console.error(chalk.red(`\nError: ${err.message}\n`));
      }

      ask();
    });
  }

  ask();
}

async function main() {
  const { model } = await inquirer.prompt([
    {
      type: 'list',
      name: 'model',
      message: 'Choose DeepSeek model',
      choices: ['deepseek-chat', 'deepseek-reasoner'],
    },
  ]);

  await chat(model);
}

main().catch((err) => {
  console.error(err);
  process.exit(1);
});

File: deepseek_gemini_cli.py
#!/usr/bin/env python3
"""Minimal Gemini-style CLI for DeepSeek models.

Supports chat and simple tool-call examples using DeepSeek's
``deepseek-chat`` and ``deepseek-reasoner`` models. The interface is
inspired by Google's Gemini CLI but routed to DeepSeek APIs.

The API key is hard-coded at the user's request. For real deployments
consider using environment variables or secret management.
"""
import argparse
import httpx

API_KEY = "sk-9af038dd3bdd46258c4a9d02850c9a6d"
API_URL = "https://api.deepseek.com/v1/chat/completions"


def call_deepseek(model: str, messages, tools=None):
    """Call DeepSeek chat completion endpoint."""
    headers = {
        "Authorization": f"Bearer {API_KEY}",
        "Content-Type": "application/json",
    }
    payload = {"model": model, "messages": messages}
    if tools:
        payload["tools"] = tools
    with httpx.Client(timeout=30.0) as client:
        resp = client.post(API_URL, headers=headers, json=payload)
        resp.raise_for_status()
    data = resp.json()
    return data["choices"][0]["message"]["content"]


def cmd_chat(args):
    messages = [{"role": "user", "content": args.prompt}]
    print(call_deepseek("deepseek-chat", messages))


def cmd_tool(args):
    messages = [{"role": "user", "content": args.prompt}]
    tools = [
        {
            "type": "function",
            "function": {
                "name": "echo",
                "description": "Echo back the supplied text",
                "parameters": {
                    "type": "object",
                    "properties": {"text": {"type": "string"}},
                    "required": ["text"],
                },
            },
        }
    ]
    print(call_deepseek("deepseek-reasoner", messages, tools=tools))


def main():
    parser = argparse.ArgumentParser(description="DeepSeek CLI")
    sub = parser.add_subparsers(dest="command")

    chat_p = sub.add_parser("chat", help="Chat with deepseek-chat model")
    chat_p.add_argument("prompt", help="Prompt to send")
    chat_p.set_defaults(func=cmd_chat)

    tool_p = sub.add_parser("tool", help="Example tool call using deepseek-reasoner")
    tool_p.add_argument("prompt", help="Prompt to send")
    tool_p.set_defaults(func=cmd_tool)

    args = parser.parse_args()
    if not args.command:
        parser.print_help()
        return
    args.func(args)


if __name__ == "__main__":
    main()
